### DISCUSSION TITLE
filters management

### DISCUSSION THREAD

**Discussion by gimame** [WinMerge/winmerge#1015] [Q&A]
Hi, I am starting to use this excellent program (in WIN10) and I would like to filter hidden system files so that the difference is not highlighted by comparing folders.
I don't understand how to do it, can some expert help me by indicating the steps to take?
Thanks.

**Comment by sdottaka** [comment]
Unfortunately, WinMerge does not have the ability to ignore hidden files or folders.

If the hidden files or folders have a specific name that distinguishes them from other regular files or folders, such as .git, then defining that name in the File filters will cause the files or folders with that name to be ignored.

There are already installed file filters that ignore folders for version control software such as git, so see if you can use it.

![image](https://user-images.githubusercontent.com/98126/138112815-2aecfdc6-52ee-4ad6-bc23-263c9ef6dde3.png)

![image](https://user-images.githubusercontent.com/98126/138112713-584b1c2a-dbc9-4de2-be4b-ac4d6d77d2ba.png)


For more information about file filters, please refer to the following

https://manual.winmerge.org/en/Filters.html

**Comment by gimame** [comment]
Thanks for the feedback, what I would like to filter is:
"desktop.ini" file
"ZbThumbnail.info" file
"System Volume Information" folder

As they are usually different, I see an unequal comparison of drives or folders forcing me to manually verify the reason.
How can I prevent this from happening?
Thanks.
  ----- Original Message -----

**Comment by sdottaka** [comment]
An example of file filters is attached.

Extract the attached file #1015.zip, install the file #1015.flt inside as follows, and compare the folders.

[#1015.zip](https://github.com/WinMerge/winmerge/files/7384844/1015.zip)


![#1015](https://user-images.githubusercontent.com/98126/138184537-b02ae33c-ebc1-473b-9744-bfe1c74496d8.gif)

**Comment by gimame** [comment]
Good help, thanks.
And I apologize for the late reply due to other commitments.

Starting from your model, I have created my own filter (see attachment) which seems to work.
Now, however, this problem arises:
I can no longer see all the comparisons even if they are identical, the options in the Message box are reset (see attachment) and in Options-Comparison mode I have selected Full contents.
What is the possible cause?
  ----- Original Message -----

**Comment by sdottaka** [comment]
I tried using the attached file filter, but could not reproduce the problem.

If you do not see the identical files in folder compare window, try looking at the check status of each menu item in the View menu.

![image](https://user-images.githubusercontent.com/98126/138558149-bddb91c8-f386-48d9-8af1-87d897a5cf15.png)

**Comment by gimame** [comment]
Hi, I have the same problem even disabling the filter.
But which version do you use?
I tried uninstalling and reinstalling the downloadable version and the View menu is different (see attachments).
Thank you.
  ----- Original Message -----

**Reply by sdottaka** [reply]
That screenshot is from WinMerge version 2.16.14 on Windows 11.

Unfortunately I am not able to see the attachment you have attached.

Please have a look at the following URL

https://github.com/WinMerge/winmerge/discussions/1015

**Reply by gimame** [reply]
Note that URL #1015 is exactly this one.
I am using the actual downloading version 2.16.14.0 X64 on W10
but menu is just different, look at the attachments.
![screen1](https://user-images.githubusercontent.com/92856318/138598109-ef15ff30-9291-46ce-b763-cb6af430b1e3.JPG)
![screen2](https://user-images.githubusercontent.com/92856318/138598110-42874a20-ff56-4259-9af7-c9df39e5e3c6.JPG)
![screen3](https://user-images.githubusercontent.com/92856318/138598111-f989389a-236b-4ca5-bf29-880358cba4da.JPG)

**Reply by sdottaka** [reply]
Try displaying the View menu when the folder comparison window is displayed.

**Reply by gimame** [reply]
OK, thanks very much, now all is good!

**Comment by gimame** [comment]
OK the attachments are on the discussion 1015.
  ----- Original Message -----

**Comment by gimame** [comment]
Dear Takashi Sawanaka, thank you very much, now everything is clear and working.
Thank you for the patience of a diligent and prompt assistance, I solved my (perhaps stupid) poor problems.
  ----- Original Message -----
